pkg/virt-operator/resource/apply/reconcile.go

```diff
@@ -129,11 +129,42 @@ func InjectPlacementMetadata(componentConfig *v1.ComponentConfig, podSpec *corev
 	if podSpec == nil {
 		podSpec = &corev1.PodSpec{}
 	}
+
 	if componentConfig == nil || componentConfig.NodePlacement == nil {
 		componentConfig = &v1.ComponentConfig{
-			NodePlacement: &v1.NodePlacement{},
+			NodePlacement: &v1.NodePlacement{
+				Affinity: &corev1.Affinity{
+					NodeAffinity: &corev1.NodeAffinity{
+						PreferredDuringSchedulingIgnoredDuringExecution: []corev1.PreferredSchedulingTerm{
+							{
+								Preference: corev1.NodeSelectorTerm{
+									MatchExpressions: []corev1.NodeSelectorRequirement{
+										{
+											Key:      "node-role.kubernetes.io/control-plane",
+											Operator: corev1.NodeSelectorOpExists,
+										},
+									},
+								},
+								Weight: 100,
+							},
+							{
+								Preference: corev1.NodeSelectorTerm{
+									MatchExpressions: []corev1.NodeSelectorRequirement{
+										{
+											Key:      "node-role.kubernetes.io/worker",
+											Operator: corev1.NodeSelectorOpDoesNotExist,
+										},
+									},
+								},
+								Weight: 99,
+							},
```

Comment:
Why do we need affinity for the workers?

Reply:
@fossedihelm raised the concern that in some environments you can have nodes that are both workers and CPs. In this case, we'd prefer running on nodes that are **just** CP, and not worker+CP. It's a small nuance that might be relevant for some environments.

The reason for the weight being slightly lower is that it's a higher priority to run on CP nodes than avoid running on worker nodes.